docs/hello_nextflow/03_hello_containers.md

```diff
@@ -1,82 +1,442 @@
 
 ```bash
-<example command>>
+cowsay "Hello World"
+```
+
+Output:
+
+```console title="Output"
+ _____________
+< Hello World >
+ -------------
+        \   ^__^
+         \  (oo)\_______
+            (__)\       )\/\
+                ||----w |
+                ||     ||
 ```
 
 This should complete immediately, and you should now see a file called `<file>` in your working directory.
 
-#### 1.4. Exit the container
+### 1.4. Exit the container
+
+To exit the container, you can type `exit` at the prompt.
 
 ```bash
 exit
 ```
 
+Your prompt should now be back to what it was before you started the container.
+
 ### Takeaway
 
 You know how to pull a container and run it interactively, and you know how to use that to try out commands without having to install any software on your system.
 
 ### What's next?
 
-Learn how to use containers from within a workflow.
+Learn how to make data from your host system available to a container.
+
+## 2. Mounting data into containers
+
+When you run a container, it is isolated from the host system by default.
+This means that the container can't access any files on the host system unless you explicitly tell it to.
+One way to do this is to **mount** a **volume** from the host system into the container.
+
+Prior to working on the next section, confirm that you are in the `hello-nextflow` directory.
+
+```bash
+cd /workspace/gitpod/nf-training/hello-nextflow
+```
+
+### 2.1. Launch the container interactively with a mounted volume
+
+```bash
+docker run --rm -it --entrypoint /bin/sh -v $(pwd)/data:/data rancher/cowsay
+```
+
+This command mounts the `data` directory in the current working directory on the host system into the `/data` directory inside the container.
+We can explore the contents of the `/data` directory inside the container:
+
+```console title="Output"
+/ # ls
+bin    data   dev    etc    home   lib    media  mnt    opt    proc   root   run    sbin   srv    sys    tmp    usr    var
+/ # ls data
+bam              greetings.csv    ref              sample_bams.txt  samplesheet.csv
+```
+
+### 2.2. Use the mounted data
+
+Now that we have mounted the `data` directory into the container, we can use the `cowsay` command to display the contents of the `greetings.csv` file.
+
+```bash
+cat data/greetings.csv | cowsay
+```
+
+Output:
+
+```console title="Output"
+ _____________________
+< Hello,Bonjour,Holà >
+ ---------------------
+        \   ^__^
+         \  (oo)\_______
+            (__)\       )\/\
+                ||----w |
+                ||     ||
+/ # exit
+```
+
+### Takeaway
+
+You know how to pull a container and run it interactively, and you know how to use that to try out commands without having to install any software on your system.
+
+### What's next?
+
+Learn how to get a container image for any pip/conda-installable tool.
 
 ---
 
-## 2. Use the container in a workflow
+## 2. Get a container image for a pip/conda-installable tool
```

Comment:
Even more generally though, this deeper review is making me realize that it feels like this section might be another slight distraction from the goal of this module, which is just to bridge the gap between Hello World and Hello Genomics (both of which use prebuilt containers that we hand to them). 

The key things we really need this module to cover are (1) running a container interactively so we can test commands, and (2) using a container from a workflow. I'd like to propose moving this current section 2 (get/build) to _after_ the section on running in a workflow. Then we can make it clear the first two sections are must-do for the training, and the latter two are take-home exercises or stretch goals, however we want to call them.